### ISSUE TITLE
[Yul] optimizer's AST copier leaks memory

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#6734]
## Description

Leak site | Avg. Leak size | Fuzz iterations until out-of-memory
------------ | -------------|---------------------------------------------
https://github.com/ethereum/solidity/blob/37375b42715efc2c131b35dad0c4c386da2703a0/libyul/optimiser/ASTCopier.cpp#L88 | 170.3 MB | 2.01 million
https://github.com/ethereum/solidity/blob/2d2f4943ded30ff3957b500f2262858e94245fd3/libyul/YulString.h#L53-L66 | 74.67 MB | 2.53 million

## Environment

- Compiler version: latest develop

**Comment by chriseth** [comment]
Are you sure this is not just the YulStringRepository leak? ASTCopier::translateVector is one of the main components in the Disambiguator, the component that probably causes most of the strings to be generated.

**Comment by bshastry** [comment]
I copied the call sites reported by the fuzzer as is. It is possible that they are related.

**Comment by axic** [comment]
@bshastry is this still the case? @chriseth could this be a contributor to the issue we're seeing in #10405 ?

**Comment by chriseth** [comment]
I don't think this is related to performance. The house-keeping of DataFlowAnalyzer is just insane.

**Comment by clonker** [comment]
We can't reproduce and/or observe issues with this anymore and if it actually is a corner case in the `YulStringRepository` I recon we will rid ourselves of it as soon as https://github.com/argotorg/solidity/pull/15969 is in. Closing this issue on grounds of that.
